continuous-integration/reference-architectures/README.md

```diff
@@ -1,3 +1,181 @@
+Know the limitations of compiled integrations and receive immediate notifications about failures.
+- [Visibility/Open Success](https://martinfowler.com/articles/continuousIntegration.html#EveryoneCanSeeWhatsHappening): 
+The state of the product is reported constantly on CI services.
+- [Testing Availability](https://martinfowler.com/articles/continuousIntegration.html#AutomateDeployment): 
+Automation allows repeatable deployment to test servers, elevating functional testing on par 
+with release.
+
+### Pipeline: A series of systems executing builds
+A [CI pipeline](https://www.redhat.com/en/topics/devops/what-cicd-pipeline) is a series of automated 
+processes, systems and triggers that execute in order to push software from raw code to a compiled 
+product that may be deployed.  
+```mermaid
+flowchart LR
+    subgraph P["<b>Pipeline</b>"]
+        subgraph DS[Development System]
+            P1
+            P2
+            P1[Commit]-->P2[Validate]
+        end
+        subgraph CIS[Continuous Integration System]
+            P3
+            P4
+            P5
+            P6
+            P7
+            P3[Compile]-->P4[Test]
+            P4-.->P5[Package]
+            P4-->P6[Report]
+            P5-->P7[Publish]
+        end
+        subgraph CDS[Continuous Deployment System]
+            P8
+            P9
+            P8[Orchestrate]-->P9[Release]
+            P9-.->P8
+        end
+        
+        DS --> |trigger| CIS
+        CIS --> |trigger| CDS
+    end
+    style P fill:#A7C7E7,stroke:#666,stroke-width:4px
+    style DS fill:#E5E4E2,stroke:#333,stroke-width:2px
+    style CIS fill:#E5E4E2,stroke:#333,stroke-width:2px
+    style CDS fill:#E5E4E2,stroke:#333,stroke-width:2px,stroke-dasharray: 5 5
+```
+The CI pipeline is composed of several systems linked together through a series of automation 
+triggers: 
+1. _Development System_  
+The sum of [code commits](https://en.wikipedia.org/wiki/Commit_(version_control)) and [code 
+versioning systems](https://en.wikipedia.org/wiki/Version_control), such as Git, comprised of systems 
+that manage raw code. Modern implementations, such as [GitHub](https://github.com/), 
+provide limited facilities to validate commits.
+2. _Continuous Integration System_  
+A collection of services that interact dynamically to perform such tasks as compilation, testing, 
+reporting and/or archiving build artifacts. CI services, such as [Jenkins](https://www.jenkins.io/) 
+or [Travis CI](https://www.travis-ci.com/), serve as management node for CI operations.
+3. _Continuous Deployment System_  
+Tooling and services, such as [Ansible](https://www.ansible.com/) or [Puppet](https://puppet.com/), 
+that deploy published software into one or more venues for extended testing or delivery. Often, 
+software outputs from the CI pipeline are deployed to development systems for extended user testing.
+
+### Build systems are composed of discrete process steps
```

Comment:
This new section title makes it seem like the contents below are unrelated to the section above, but that is not the case.

Reply:
If each heading title is related, maybe add a numbering so its more clear for the reader?

Reply:
I changed the wording to reference the CI Pipeline which is what is really being discussed. This will be in the next commit.